### ISSUE TITLE
Blueprint FAQ

### ISSUE THREAD

**Issue by nzrakic** [chmjs/chameleon-docs#7]
Please post your questions in new comments.
I'll turn this into a markdown document once we finish gathering questions.

**Comment by nzrakic** [comment]
**Q/A:**
* What does a *working model* represent?
  * A non-compiled schema,
  * Work in progress model, if you will.
* What constitutes a *working model*?
  * A data package with it's records, fields and view models:
    * A single data package may have one or more records,
      * A single record may have one or more fields,
    * One default view model per record,
    * Zero or more custom view models per record.
* Can I have more than one working model checked out at any time?
  * Yes, of course. You can create multiple data packages.
  * Each data package represents a separate working model.
* What field types are supported?
  * Client may create fields of the following types: `bool`, `text`, `number`, `date`, `geopoint`, `relation`.
  * Additionally, `primary` field type is always created for each new record.
* What field value validations are available?
  * `required` (bool),
  * `regex` (string),
  * `minValue` (string),
  * `maxValue` (string),
  * `minLength` (integer),
  * `maxLength` (integer)
  * Additionally, `unique` (bool) will be added soon.
* Which validations are available per field type?
  * all types support: `required`
  * `text` supports: `regex`, `minLength`, `maxLength`
  * `number` supports: `regex`, `minValue`, `maxValue`
  * `date` supports: `minValue`, `maxValue`
* Is it possible to check out a *working model* from a provided schema?
  * Yes, a working model will be created from the provided schema. It will **not** be compiled nor committed.
  * Temporary IDs must be used when referencing records and fields in relational fields or view model fields.
* What is a *compiled* schema?
  * A complete JSON-formatted schema created from working model elements: data package, records, fields and view models.
* What is the *latest* schema version?
  * A non-committed (not saved) but compiled schema of the current working model.
* What is a *committed* schema version?
  * A compiled schema that is stored in schema version history,
  * A _new version of a data package_, if you will.
* How can a committed schema version be used in a space?
  * First it has to be installed into that space,
  * All it's dependencies (schema versions it depends on) must be installed as well (recursively).

**Comment by nzrakic** [comment]
@chmjs/tools Please comment if you have something to add to the above list.

**Comment by manico** [comment]
Just please organise in docs as question and answer. For example:

**What does a working model represent?**
Working model represents a non-compiled schema, a model which is work in progress.

**Comment by nzrakic** [comment]
Here it is: [FAQ](https://chameleon-docs.nsoft.com/developers/storage/blueprint/FAQ.html).

**Comment by bernardmo** [comment]
Can we create relations between 2 working models?

**Comment by nzrakic** [comment]
> Can we create relations between 2 working models?
No. In order to use an external reference (a record from some other data package) in your working model it first needs to be compiled into a new schema version, committed into schema version history and installed into your space.

**Comment by Lanchi** [comment]
- What does `versionId` present in data package schema versions lists?
- Does `versionId` change after schema compile and schema installation actions?
- How is latest schema generated on uncompiled working model?
- After schema is installed, will latest schema version be identical (same records/views and version id)?
- Can schema be installed without previous compile step?

**Comment by nzrakic** [comment]
>  What does versionId present in data package schema versions lists?

Added to the FAQ:

Unique identifier of the compiled schema version. It is calculated from the following properties:
* `spaceId`
* `dataPackageId`
* `ordinal`

Ordinal number of the compiled schema version. It is incremented for each new schema version.

>   Does versionId change after schema compile and schema installation actions?

Added to the FAQ:

No.

>  How is latest schema generated on uncompiled working model?

Already covered in the FAQ under following question:

**What is the *uncommitted* schema version?**

>  After schema is installed, will latest schema version be identical (same records/views and version id)?

Let me rephrase your question:

*After schema is committed, will uncommitted schema version be identical (same records/views and version id)?*

Only the `versionId` will change. The rest of the schema version will remain the same until changes are made to the working model.


>   Can schema be installed without previous compile step?

Already covered in the FAQ under following question:

**How can a committed schema version be used in a space?**
